perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/UsersManagerBlImpl.java

```diff
@@ -475,13 +477,13 @@ public void deleteUser(PerunSession sess, User user) throws RelationExistsExcept
 	public void deleteUser(PerunSession sess, User user, boolean forceDelete) throws RelationExistsException, MemberAlreadyRemovedException, UserAlreadyRemovedException, SpecificUserAlreadyRemovedException {
 		try {
 			this.deleteUser(sess, user, forceDelete, false);
-		} catch (AnonymizationNotSupportedException ex) {
+		} catch (AnonymizationNotSupportedException | ConsentNotExistsException | UserNotExistsException ex) {
 			//this shouldn't happen with 'anonymizedInstead' set to false
 			throw new InternalErrorException(ex);
 		}
 	}
 
-	private void deleteUser(PerunSession sess, User user, boolean forceDelete, boolean anonymizeInstead) throws RelationExistsException, MemberAlreadyRemovedException, UserAlreadyRemovedException, SpecificUserAlreadyRemovedException, AnonymizationNotSupportedException {
+	private void deleteUser(PerunSession sess, User user, boolean forceDelete, boolean anonymizeInstead) throws RelationExistsException, MemberAlreadyRemovedException, UserAlreadyRemovedException, SpecificUserAlreadyRemovedException, AnonymizationNotSupportedException, ConsentNotExistsException, UserNotExistsException {
```

Comment:
I believe UserNotExistsException won't be thrown and should be omitted from the signature.